libraries/WiFi/src/WiFiAP.cpp

```diff
@@ -432,14 +204,7 @@ bool WiFiAPClass::softAPenableIPv6(bool enable)
 
 IPAddress WiFiAPClass::softAPIPv6()
 {
-    static esp_ip6_addr_t addr;
-    if(WiFiGenericClass::getMode() == WIFI_MODE_NULL){
-        return IPAddress(IPv6);
-    }
-
-    if(esp_netif_get_ip6_linklocal(get_esp_interface_netif(ESP_IF_WIFI_STA), &addr)){
-        return IPAddress(IPv6);
-    }
-    return IPAddress(IPv6, (const uint8_t *)addr.addr, addr.zone);
+    return AP.localIPv6();
```

Comment:
If this is a link-local address, then it is probably not the best solution for an access point.  Link-local addresses require a scope suffix and have known issues with access via a browser (Windows happens to work without a suffix if it is the default link; other OS do not work).

For IPv6, an AP should assign a Unique Local Address range (ULA), which is the equivalent of a private address range, and then use Router Advertisement (RA) to assign addresses to clients (similar to DHCP).

You can use either random ULAs and access point addresses, or set known static short values, e.g. set fc00::1 as the static address for the AP (similar to setting 192.168.0.1 as a static IPv4 AP).

Reply:
i.e. Rather than AP.linkLocalIPv6() this should return `AP.globalIPv6()`, which is similar to IPv4 (IPv4 localIP() returns a global IPv4 address, for the local (vs remote) node).

Reply:
I have no issue with link local addresses here. Running on Mac and Linux things (also ESPs). The AP provides LinkLocal IPv6 and this is what is returned here. I'm renaming it to show that. As for Global, I was not able to get that going on my ESP

Reply:
Happy with the appropriate name; globalIPv6() can come later.

For an IPv6 AP there is quite a bit of set up, similar to setting up DHCPS for IPv4, static address, etc. Except IPv6 uses RA as the primary mechanism.  If all you want is to connect to the device over WiFi, then plain RA is all you need; for an actual router (or host name usage) you would need a way to also provide DNS.  

I don't even know what is possible here in ESP-IDF, but definitely a separate thing.